setup.sh

```diff
@@ -19,48 +13,112 @@ error() { echo -e >&2 "\e[31m[!] ${1:-}\e[0m"; exit 1; }
 [[ "${EUID}" -eq  0 ]] && warning "Script running as root, you should run this as an unprivileged user."
 
 usage() {
-    echo "Usage: $0 [-a ARCH] [-l LICENSE_KEY] [-v VERSION]"
+    echo "Usage: $0 "
+    echo -e "\t -a ARCH\t optional\t Specify architecture (x64/armn/arm64)"
+    echo -e "\t -l LICENSE_KEY\t optional\t Specify license key for install"
+    echo -e "\t -v VERSION\t optional\t Specify version to install"
+    echo -e "\t -u \t\t optional \t Use unstable channel for package repositories"
+    echo -e "\t -h \t\t optional \t Display this message"
     [[ "${1:-}" == "error" ]] && exit 1
     exit 0
 }
 
 quick_start() {
     info "Installation complete."
     echo -e "\n\e[1mLearn how to use Enclave at https://enclave.io/docs/\e[0m"
-    echo "Quick start:"
+    echo -e "Quick start:"
     echo -e "    \e[1menclave add [PEER_NAME] -d [DESCRIPTION]\e[0m to authorise a connection to another system running enclave."
     echo -e "    \e[1menclave status\e[0m for status.\n"
 }
 
-install_dependencies() {
-    info "Checking/installing dependencies."
-    deps=(tar wget)
-    # Install common and distro specific deps
+get_distro_family() {
     if grep -qi "rhel" /etc/os-release; then
-        deps+=(libicu)
-        sudo yum install -y "${deps[@]}"
-    elif grep -qi "debian" /etc/os-release; then
-        info "Updating package index."
-        sudo apt-get update -qq
-        # Different versions of Debian/Ubuntu ship different versions of libicu
-        deps+=("$(apt-cache search -n "^libicu[0-9]+$" | cut -d" " -f1)")
-        # RaspberryPi OS/Raspbian needs libsodium-dev
-        if grep -qi "raspbian" /etc/os-release; then 
-            deps+=("libsodium-dev")
-        fi 
-        sudo apt-get install -yq "${deps[@]}"
+        echo "rhel"
+    elif grep -qi "raspbian" /etc/os-release; then
+        echo "raspbian"
+    elif grep -qi "debian" /etc/os-release || grep -qi "ubuntu" /etc/os-release; then
+        echo "debian"
     elif grep -qi "arch" /etc/os-release; then
-        deps+=(icu libsodium)
-        sudo pacman -Syq --noconfirm --needed "${deps[@]}"
+        echo "arch"
+    fi
+}
+
+install_apt_package() {
+    # Install the pre-requisites for an apt install
+    info "Updating package index."
+    sudo apt-get update -qq
+    deps=(apt-transport-https wget)
+    sudo apt-get install -yq "${deps[@]}"
+
+    # Add and trust the Enclave package repository
+    info "Adding Enclave GPG package signing key."
+    wget -qO- https://packages.enclave.io/apt/pubkey.gpg | sudo apt-key add - >/dev/null 2>&1
+    info "Adding the Enclave package repository."
+    # shellcheck disable=SC2024
+    wget -qO- "https://packages.enclave.io/apt/${ENCLAVE_PKG_LIST}" | sudo tee "/etc/apt/sources.list.d/${ENCLAVE_PKG_LIST}" >/dev//null
```

Comment:
Is the double-forward slash required?

Reply:
Not, that's a typo. One of those weird ones that will actually still work though!